util/normalization_functions.R

```diff
@@ -1001,6 +1001,162 @@ CNProcessing <-  function(array.dt, seq.dt,
   
 }
 
+
+SeuratIntegration <-  function(array.dt, seq.dt,
+                               n_dims = 50,
+                               vbose = TRUE) {
+  
+  # This function takes array and RNA-seq data in the form of data.tables
+  # to be normalized and concatenated through Seurat integration.
+  #
+  # Args:
+  #   array.dt: data.table of array data where the first column contains
+  #             gene identifiers, the columns are samples,
+  #             rows are gene measurements
+  #   seq.dt:   data.table of RNA-seq data where the first column contains
+  #             gene identifiers, the columns are samples,
+  #             rows are gene measurements
+  #   n_dims:   number of dimensions for reduction (default: 50)
+  #   vbose:    use verbose screen output (default: TRUE, may give error if FALSE)
+  #
+  # Returns:
+  #   array_seq.integrated: an integrated Seurat object including array and seq
+  
+  require(data.table)
+  # Error-handling
+  array.is.dt <- "data.table" %in% class(array.dt)
+  seq.is.dt <- "data.table" %in% class(seq.dt)
+  any.not.dt <- !(any(c(array.is.dt, seq.is.dt)))
+  if (any.not.dt) {
+    stop("array.dt and seq.dt must both be data.tables")
+  }
+  if (!(all(array.dt[[1]] %in% seq.dt[[1]]))) {
+    stop("Gene identifiers in data.tables must match")
+  }
+  
+  array.dt <- ensure_numeric_gex(array.dt)
+  seq.dt <- ensure_numeric_gex(seq.dt)
+  
+  # List of seurat objects to be integrated
+  array_seq_list <- list(array = array.dt[,-1],
+                         seq = seq.dt[,-1])
+  
+  array_seq_list <- lapply(X = array_seq_list,
+                           FUN = Seurat::CreateSeuratObject)
+  
+  # SCTransform
+  array_seq_list <- lapply(X = array_seq_list,
+                           FUN = Seurat::SCTransform,
+                           verbose = vbose)
+  
+  features <- Seurat::SelectIntegrationFeatures(object.list = array_seq_list,
```

Comment:
Please add a comment about what this is doing. I'd probably include the number of features getting selected by default, but it might be better to explicitly set `nfeatures` here using the default value.